core/app/models/spree/promotion.rb

```diff
@@ -91,23 +91,28 @@ def eligible_rules(promotable, options = {})
       # Promotions without rules are eligible by default.
       return [] if rules.none?
       eligible = lambda { |r| r.eligible?(promotable, options) }
-      specific_rules = rules.for(promotable)
+      specific_rules = rules.select { |rule| rule.applicable?(promotable) }
```

Comment:
Since you're no longer using PromotionRule.for here I believe it can just be removed as this seems to be the only place it's used.